tests/test_dragon_launcher.py

```diff
@@ -191,27 +192,28 @@ def test_dragon_connect_bind_address(monkeypatch: pytest.MonkeyPatch, test_dir:
         ctx.setenv("SMARTSIM_DRAGON_SERVER_PATH", test_dir)
         # avoid finding real interface
         ctx.setattr(
-            "smartsim._core.launcher.dragon.dragonLauncher.get_best_interface_and_address",
+            "smartsim._core.launcher.dragon.dragonConnector.get_best_interface_and_address",
             lambda: IFConfig(interface="faux_interface", address="127.0.0.1"),
         )
         # we need to set the socket value or is_connected returns False
         ctx.setattr(
-            "smartsim._core.launcher.dragon.dragonLauncher.DragonLauncher._handshake",
-            lambda self, address: setattr(self, "_dragon_head_socket", mock_socket),
+            "smartsim._core.launcher.dragon.dragonLauncher.DragonConnector._handshake",
+            lambda self, address: ...,
         )
         # avoid starting a real authenticator thread
         ctx.setattr("zmq.auth.thread.ThreadAuthenticator", MockAuthenticator)
         # avoid starting a real zmq socket
         ctx.setattr("zmq.Context.socket", mock_socket)
-        # avoid starting a real process for dragon entrypoint
         ctx.setattr("subprocess.Popen", lambda *args, **kwargs: MockPopen())
 
-        dragon_launcher = DragonLauncher()
-        dragon_launcher.connect_to_dragon(test_dir)
+        dragon_connector = DragonConnector()
+        dragon_connector.connect_to_dragon()
 
         chosen_port = int(mock_socket.bind_address.split(":")[-1])
         assert chosen_port >= 5995
 
+        dragon_connector._authenticator.stop()
```

Comment:
nice catch. were the tests hanging?

Reply:
Only because I changed how the connector works vs how the launcher used to work.